lib/ansible/plugins/strategy/__init__.py

```diff
@@ -318,6 +318,14 @@ def _queue_task(self, host, task, task_vars, play_context):
                 self._cur_worker += 1
                 if self._cur_worker >= len(self._workers):
                     self._cur_worker = 0
+                if len(task.forks) > 0 and task._parent._play.strategy != "free":
+                    if task.run_once:
+                        display.warning("Ignoring 'forks' as 'run_once' is also set for '%s'" % task.get_name())
+                    else:
```

Comment:
we might want to warn 'forks is ignored due to strategy == free'

Reply:
hmm, its implemented in free strat, unsure why you need to check for free here then

Reply:
The code in StrategyBase limits the amount of workers, which is the proper solution for linear. For the free strategy the workers should not get limited as all the other workers might be used for other tasks.

Reply:
Please have a look at the third patch for forks support with the free strategy.